src/db-service.ts

```diff
@@ -26,15 +26,29 @@ export class ItemTagService {
     sql`, `
   );
 
+  private static allColumnsForJoins = sql.join(
+    [
+      [['item_tag', 'id'], ['id']],
+      [['item_tag', 'tag_id'], ['tagId']],
+      [['item_tag', 'item_path'], ['itemPath']],
+      [['item_tag', 'creator'], ['creator']],
+      [['item_tag', 'created_at'], ['createdAt']],
+    ].map(c => sql.join(c.map(cwa => sql.identifier(cwa)), sql` AS `)),
+    sql`, `
+  );
+
   /**
    * Get all item tags, local or inherited, for the given `item`.
    * @param item Item
    * @param transactionHandler Database transaction handler
    */
   async getAll(item: Item, transactionHandler: TrxHandler): Promise<ItemTag[]> {
     return transactionHandler.query<ItemTag>(sql`
-        SELECT ${ItemTagService.allColumns} FROM item_tag
-        WHERE item_path @> ${item.path}
+        SELECT ${ItemTagService.allColumnsForJoins} FROM item_tag
+        INNER JOIN tag ON tag.id = tag_id 
```

Comment:
I noticed this inner join, won't it affect other calls? (I fear that it blows up compose since it's still using both item tags and tags). I don't see a case where you would like to have the item tags but not the corresponding tags. But please check it does not blows compose (for public, item login/pseudonymized) otherwise apply changes there.